src/main/java/org/venice/beachfront/bfapi/model/oauth/GeoAxisCommonName.java

```diff
@@ -0,0 +1,67 @@
+package org.venice.beachfront.bfapi.model.oauth;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.TreeNode;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+
+public interface GeoAxisCommonName {
+	public String toString();
+	
+	public static class Deserializer extends JsonDeserializer<GeoAxisCommonName> {
+		@Override
+		public GeoAxisCommonName deserialize(JsonParser p, DeserializationContext ctxt) throws IOException, JsonProcessingException {
+			TreeNode node = p.getCodec().readTree(p);
+			if (node.isValueNode() && ((JsonNode)node).isTextual()) {
+				return new SingleString(((JsonNode)node).asText());
+			}
+			if (node.isArray()) {
+				LinkedList<String> cns = new LinkedList<>();
+				for (int i=0; i < node.size(); i++) {
+					if (!node.get(i).isValueNode()) {
```

Comment:
So the only thing the array can hold is Strings, and we're sure it can't be anything else at this point?

Reply:
I think that's safe to say